### ISSUE TITLE
Game type boilerplate

### ISSUE THREAD

**Issue by GabeAtWork** [simiancraft/create-phaser-app#37]
At the moment, the demo game is a platformer, a game that uses 2D physics with a player controlled entity that moves around. This may not be helpful to someone who wants to make a strategy game, or a management game, or a card game, ...

So I was wondering, what are the plans (if any) to support different game type demos in the phaser app? I can imagine maybe a RTS, a management game, a Zelda-like isometric one...

From a usability standpoint, I imagine there could be several scenes with different demos available. But if you plan on doing such a thing with create-phaser-app, I'm wondering how to structure the app so that someone can easily identify the parts they need for their type of game and discard the rest, while keeping the opinionated file structure that makes getting right into development easily.

**Comment by the-simian** [comment]
I was thinking if using something like inquirer.js someday to handle the initial scaffolding, I think we could take feedback then and change how the project scaffolds.

I'm open to adding more 'game types', but I wanted to start with a sidescroller;  since it seems to be a common enough starting point. Maybe an overhead view after that? (like zelda). A lot of the things I'm working on now would be reused, for instance using a FSM to manage player control is pretty much the same on all of them.

Basically, yes there's plans for this, and it will be easier and easier to add more game times as we move along.

As always I appreciate your input!

**Comment by GabeAtWork** [comment]
Cool! I'll be interested on working on the zelda-like game, and later if I find out how to make it, a RTS or turned base strategy game. It requires me to learn along, but we'll see.

I'll look into inquirer for scaffolding. I know it's probably not the most urgent thing, but at least it's something I understand well enough to make a hopefully 'good enough' PR for ^^

**Comment by the-simian** [comment]
Yeah, another thing that @vantreeseba and I have kicked around is making it so that at the beginning of the game you can level select, and the levels demonstrate some differnt, but common game types. So there's obviously some asset swapping (probably not menus, depends). Sounds and music can stay the same, and the underlying tooling, like the FSM for behavior management, behavior tree tooling, can more-or-less be reused.

I figure its best to get this 'game' solid, and something that's a very compelling example to learn from, and branch from there. I know you mentioned a zelda game, and I 100% want to make a top-down zelda-type the very next priority, especially since it reuses so many systems.

Appreciate the discussion.

**Comment by the-simian** [comment]
@vantreeseba and I came up with a cool idea for this.

We want
- A top down game (like zelda)
- a sidescroller (like contra ..which we kind of have atm)
- A 'dialog driven' thing with dialog trees (visual novels)
- a *very simple* sidescroller or top scroller not much more complicated than mario.

So here is the big idea. We include these as additional views and minigames in the 'big game' (signal Overgrowth); however when you scaffold you can choose to only scaffold the 'mini game'.

So the contra type game is the main game, the top down is a separate view for the game (an overworld map), the 'simple mario-esque game' is a mini game where you can hack things (or get out of your mech in a detachable rover).

However you can scaffold *only* one of these smaller experiences as well.

So the scaffold dialog might say like

- [ ] Scaffold a Dialog game (like a visual novel),
- [ ] Scaffold a simple sidescroller (like mario)
- [ ] Scaffold a complex sidescroller ( like contra/ megamanX)
- [ ] Scaffold a top-down game (like zelda)
- [x] Scaffold The Signal Overgrowth (all of the above in the context of a larger project)
